test/test_in_toto_record.py

```diff
@@ -43,101 +41,135 @@ class TestInTotoRecordTool(unittest.TestCase):
       in_toto_record_main()
-    self.assertLessEqual(logging.getLogger().getEffectiveLevel(), logging.INFO)
-    # Reset log level
-    logging.getLogger().setLevel(original_log_level)
-
-  def test_in_toto_record_start_stop(self):
-    """in_toto_record_start/stop run through. """
-    in_toto_record_start("test-step", self.key, [self.test_artifact])
-    in_toto_record_stop("test-step", self.key, [self.test_artifact])
-
-  def test_in_toto_record_start_bad_key_error_exit(self):
-    """Error exit in_toto_record_start with bad key. """
-    with self.assertRaises(SystemExit):
-      in_toto_record_start("test-step", "bad-key", [self.test_artifact])
-
-  def test_in_toto_record_stop_missing_unfinished_link_exit(self):
-    """Error exit in_toto_record_stop with missing unfinished link file. """
-    with self.assertRaises(SystemExit):
-      in_toto_record_stop("test-step", self.key, [self.test_artifact])
+    self.assertEqual(raise_ctx.exception.code, status)
+
+
+  def test_start_stop(self):
+    """Test CLI command record start/stop with various arguments. """
+
+    # Start/stop recording
+    args = ["--step-name", "test1", "--key", self.key_path]
+    self._test_cli_sys_exit(["start"] + args, 0)
+    self._test_cli_sys_exit(["stop"] + args, 0)
+
+    # Start/stop with recording one artifact
+    args = ["--step-name", "test2", "--key", self.key_path]
+    self._test_cli_sys_exit(["start"] + args + ["--materials",
+        self.test_artifact1], 0)
+    self._test_cli_sys_exit(["stop"] + args + ["--products",
+        self.test_artifact1], 0)
+
+    # Start/stop with recording multiple artifacts
+    args = ["--step-name", "test3", "--key", self.key_path]
+    self._test_cli_sys_exit(["start"] + args + ["--materials",
+        self.test_artifact1, self.test_artifact2], 0)
+    self._test_cli_sys_exit(["stop"] + args + ["--products",
+        self.test_artifact2, self.test_artifact2], 0)
+
+    # Start/stop recording verbosely
+    args = ["--step-name", "test4", "--key", self.key_path, "--verbose"]
+    self._test_cli_sys_exit(["start"] + args, 0)
+    self._test_cli_sys_exit(["stop"] + args, 0)
+
+    # Start/stop sign with specified gpg keyid
+    args = ["--step-name", "test5", "--gpg", self.gpg_keyid, "--gpg-home",
+        self.gnupg_home]
+    self._test_cli_sys_exit(["start"] + args, 0)
+    self._test_cli_sys_exit(["stop"] + args, 0)
+
+    # Start/stop sign with default gpg keyid
+    # Default key signing only works on fully supported gpg versions
+    args = ["--step-name", "test6", "--gpg", "--gpg-home", self.gnupg_home]
+    if in_toto.gpg.util.is_version_fully_supported():
+      self._test_cli_sys_exit(["start"] + args, 0)
+      self._test_cli_sys_exit(["stop"] + args, 0)
+
+    else:
+      # Fail (exit 1) start recording on not fully supported gpg version
+      self._test_cli_sys_exit(["start"] + args, 1)
+
+      # NOTE: Stop recording with default gpg on not fully supported gpg
```

Comment:
I don't think this comment is clear enough...

Reply:
Ok, I'll reword.